lib/pubsub/subscription.js

```diff
@@ -235,9 +242,29 @@ function Subscription(pubsub, options) {
     config.methods.create = true;
   }
 
-  ServiceObject.call(this, config);
+  GrpcServiceObject.call(this, config);
   events.EventEmitter.call(this);
 
+  this.autoAck = is.boolean(options.autoAck) ? options.autoAck : false;
+  this.closed = true;
+  this.encoding = options.encoding || 'utf-8';
+  this.inProgressAckIds = {};
+  this.interval = is.number(options.interval) ? options.interval : 10;
+  this.maxInProgress =
+    is.number(options.maxInProgress) ? options.maxInProgress : Infinity;
+  this.messageListeners = 0;
+  this.paused = false;
+
+  if (is.number(options.timeout)) {
+    this.timeout = options.timeout;
+  } else {
+    // The default timeout used in gcloud-node is 60s, but a pull request times
+    // out around 90 seconds. Allow an extra couple of seconds to give the API a
+    // chance to respond on its own before terminating the connection.
+    var PUBSUB_API_TIMEOUT = 90000;
```

Comment:
This seems oddly placed, I think a better location would be outside of the constructor.